plugins/experimental/uri_signing/normalize.c

```diff
@@ -0,0 +1,382 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "normalize.h"
+#include "common.h"
+#include <string.h>
+#include <ctype.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+/* Remove Dot Algorithm outlined in RFC3986 section 5.2.4
+ * Function writes normalizes path and writes to ret_buffer */
+int
+remove_dot_segments(const char *path, int path_ct, char *ret_buffer, int buff_ct)
```

Comment:
@dsouza93 I get the following warning when i build, try adding '#include "ts/ts.h" on line 30 after '#else' in common.h:

In file included from experimental/uri_signing/normalize.c:20:0:
experimental/uri_signing/normalize.c: In function 'remove_dot_segments':
experimental/uri_signing/common.h:31:26: warning: implicit declaration of function 'TSDebug' [-Wimplicit-function-declaration]
 #define PluginDebug(...) TSDebug("uri_signing", PLUGIN_NAME " " __VA_ARGS__)
                          ^
experimental/uri_signing/normalize.c:33:5: note: in expansion of macro 'PluginDebug'
     PluginDebug("Path buffer not large enough");
     ^~~~~~~~~~~
experimental/uri_signing/normalize.c: At top level:
cc1: warning: unrecognized command line option '-Wno-cast-function-type'

Reply:
Should be resolved now

Reply:
looks good